ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -1515,19 +1542,22 @@ private void updateIdealStateWithNewPartitionsForFullAuto(String dcName,
     while (round >= 0) {
       allPartitionsDontHaveResources = true;
       for (Map.Entry<String, Set<String>> ent : partitionsToInstancesInDc.entrySet()) {
+        int numReplica = ent.getValue().size();
         Set<String> instanceNames = ent.getValue();
         Set<Integer> resources = instanceNames.stream()
-            .map(in -> instanceNameToOneResource.get(in))
+            .map(in -> instanceNameToOneResource.get(in + "_" + numReplica))
             .filter(Objects::nonNull)
             .collect(Collectors.toSet());
         if (resources.isEmpty()) {
           // Partition doesn't belong to any host that are already in the helix
         } else if (resources.size() == 1) {
+          info("[{}] Partition {} belongs to resource {} at round {}", dcName, ent.getKey(),
+              resources.iterator().next(), round);
           allPartitionsDontHaveResources = false;
           int resourceId = resources.iterator().next();
-          resourceIdToInstances.get(resourceId).addAll(instanceNames);
           // Some instance names might not have a resource name, now assign the resource name to it
-          instanceNames.forEach(in -> instanceNameToOneResource.put(in, resourceId));
+          instanceNames.forEach(in -> instanceNameToOneResource.put(in + "_" + numReplica, resourceId));
```

Comment:
Is there any hard rule or naming convention for the "resource"?

Reply:
resource names are going to be integer numbers.